### PR TITLE
fix: revolut transaction fee calculation

### PR DESCRIPTION
**Pull request by Zahrun**

### CODE REVIEW

File: src/parsers/trades/revolut/index.ts [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default async function processData(importDetails: IImport): Promise<ITrad
         tradeToAdd.ID = createID(tradeToAdd);
         tradeToAdd.exchangeID = tradeToAdd.ID;
         tradeToAdd.transactionFeeCurrency = trade['Base currency'];
-        tradeToAdd.transactionFee = parseFloat(trade.Fee);
+        tradeToAdd.transactionFee = Math.round((parseFloat(trade['Fiat amount (inc. fees)'])-parseFloat(trade['Fiat amount']))*100)/100;
```
Comment (starsoccer) [inline src/parsers/trades/revolut/index.ts:52]: Im not sure I understand why we are doing all of this. From the CSV you shared before it seems like we should only need to get the trade.Fee column.

Also its worth noting this field is for the transactionFee(not trade fee). Trade fees are just included in the trade typically

Comment (Zahrun) [reply src/parsers/trades/revolut/index.ts:52]: There are inconsistencies in the revolut export where the fee is more than the (fiat amount) - (fiat amount (inc. fee)). I checked the transactions in the revolut app and the correct amount is the one I modified the code to.
Example here 0.98 + 0.03 is more than 1.
image

Comment (starsoccer) [reply src/parsers/trades/revolut/index.ts:52]: Okay so first transaction fee is not for trade fees. Transaction fee is for the fee used to deposit/withdraw funds from an exchange as part of a trade.

Exchange fees are usually included in the rate of a trade in order to spread it across all purchased/sold units. So in this case where is the correct amount bought and fee charged?
Based on the data in that csv we have 3 values to work with:
- Fiat amount(0.98)
- Fiat amount(inc. fees) (1)
- Fee(0.03)

So if the total amount of fiat you received is 0.98 then we should be able to just subtract fiat amount(inc. Fees) from Fiat amount and use that as the true fee. We can/should also ask the revolut folks to fix this as clearly there exports are broken and rounding values

Comment (Zahrun) [reply src/parsers/trades/revolut/index.ts:52]: Thank you for clarifying what transaction fee is for, that will help me to create the other exchanges support better.
Actually, back when I used Revolut, it only supported trades from fiat to crypto, not between crypto. 
If you look at the example file I sent, it has only trades from or to EUR.
I don’t know if today it supports crypto to crypto trade, but I don’t have any example of that.

In that example, I was charged 1.00 EUR for 0.98 EUR value of crypto.

Will send a message to the Revolut team

Comment (starsoccer) [reply src/parsers/trades/revolut/index.ts:52]: Okay so then my suggestion would be to just disregard the fee field entirely and just do fiat amount minus fiat amount(inc. Fees) and then round to 2 decimals. Maybe something like this?
```
Math.floor(trade['Fiat amount (inc. fees)'] - trade['Fiat amount'])
```

Comment (Zahrun) [reply src/parsers/trades/revolut/index.ts:52]: They seem to have agreed there is an issue and they will fix it.
In the meantime, what I proposed in the PR is et round to 2 decimals with Math.round.
I think Math.floor truncates instead.

Comment (starsoccer) [reply src/parsers/trades/revolut/index.ts:52]: Well floor will always round down which I guess may be the same thing as just cutting off anything after 2 decimals. I am find with using math.round instead but I dont really see the need for us to do the multiplying by 100 and then dividing again

Comment (Zahrun) [reply src/parsers/trades/revolut/index.ts:52]: Because Round rounds to the integer. I want to round to the cent

Comment (starsoccer) [reply src/parsers/trades/revolut/index.ts:52]: Okay approving for now then but we should revisist then when they fix the export

Comment (Zahrun) [reply src/parsers/trades/revolut/index.ts:52]: Sure. Will keep you updated.

**Pull request merged**